modules/stochastic_tools/doc/content/modules/stochastic_tools/install_pytorch.md

```diff
@@ -0,0 +1,176 @@
+# Install MOOSE with Pytorch C++ API (libtorch)
+
+The way one enables pytorch capabilities in MOOSE depends on
+the operating system (Linux or Mac) and if we use HPC or just a local workstation.
+
+!alert! note
+Before we review the main approaches, it is important to emphasize that
+linking MOOSE with libtorch on Linux machines is not support if the compiler stack has been built
+using a `libc` version below 2.27 (for `libtorch v 1.8+`)
+or 2.23 (for `libtorch v1.4-1.8`). Furthermore, we do not support `libtorch` versions below
+v1.4. To check your currently used libc version use the following command:
+
+```bash
+ldd --version
+```
+
+!alert-end!
+
+## Installation on a local Mac workstation
+
+The first step of this process is to set up the a suitable conda environment. For this follow
+the instructions [here](https://mooseframework.inl.gov/getting_started/installation/conda.html).
+Next, activate the conda environment (assume `moose-torch`), clone the repository
+ and navigate to the base folder of the
+stochastic tools module (let's say it is in `~/projects`):
+
+```bash
+conda activate moose-torch
+git clone https://github.com/grmnptr/moose.git
+cd moose
+git checkout torch-compile-19571
+cd ~/projects/moose/modules/stochastic_tools
+```
+
+As a next step, we download a precompiled version of `libtorch` which is suitable
+for our system. This is carried out by a bash script as:
+
+```bash
+./scripts/setup_libtorch.sh
+```
+
+!alert! note
+The desired version of libtorch can be set by the following
+argument:
+
+```bash
+./scripts/setup_libtorch.sh --version=1.8
+```
+
+!alert-end!
+
+which creates a `libtorch` folder in the stochastic tools root directory. This folder
+will contain the necessary header files and shared object files which can be used
+for dynamic linking.
+
+The last step is to compile our MOOSE with libtorch. For this, a non-unity build
+is used (which is considerably slower than the default unity build) in the
+following manner:
+
+```bash
+MOOSE_UNITY=false make -j 8
+```
```

Comment:
Is there a way we can automatically do this in the mk file? Knowing me, I would constantly forget to do this.

Reply:
It works with unity now!